src/clib/lib/job_queue/torque_driver.cpp

```diff
@@ -530,9 +534,11 @@ torque_driver_get_qstat_status(torque_driver_type *driver,
     job_status_type status = JOB_QUEUE_STATUS_FAILURE;
 
     {
-        const char **argv = (const char **)util_calloc(2, sizeof *argv);
-        argv[0] = driver->qstat_opts;
-        argv[1] = jobnr_char;
+        char const *hard_coded_qstat_option = "-f";
+        const char **argv = (const char **)util_calloc(3, sizeof *argv);
```

Comment:
Nitpicks (I realise this isn't new code but still)

Allocate `argv` on the stack.
```suggestion
        const char *argv[3];
```

Alternatively, using fancy C++: (add `#include <vector>` at the top if it doesn't exist)
```c++
    std::vector<const char *> argv{
        hard_coded_qstat_option,
        driver->qstat_opts,
        jobnr_char,
    };
```

Then:
* Remove `free(argv);` on line 562.
* Use `argv.size()` instead of the literal `3`.
* Use `argv.data()` instead of `(const char **)argv` (the cast is unnecessary anyway; the type is already `const char **`)

You can also replace `std::vector` with `std::array` to be invited to the secret cult of C++11. (The difference is marginal and `std::array` is likely slightly harder to use, but would be the most "efficient" thing to do in this case)

Reply:
Thanks, all nitpicks implemented!